src/Core/src/VisualDiagnostics/VisualDiagnosticsOverlay.iOS.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using System.Collections.Generic;
+using Foundation;
+
+namespace Microsoft.Maui
+{
+	/// <summary>
+	/// Visual Diagnostics Overlay.
+	/// </summary>
+	public partial class VisualDiagnosticsOverlay
+	{
+		const string ScrollViewContentOffsetKey = "contentOffset";
+
+		readonly Dictionary<IScrollView, IDisposable> _scrollViews = new();
+
+		public void AddScrollableElementHandler(IScrollView scrollBar)
+		{
+			var nativeScroll = scrollBar.GetNative(true);
+			if (nativeScroll != null)
+			{
+				var dispose = nativeScroll.AddObserver(ScrollViewContentOffsetKey, NSKeyValueObservingOptions.New, FrameAction);
```

Comment:
Don't we need to remove this observer?

Reply:
It should be disposed at https://github.com/dotnet/maui/blob/85ea2218637a8d2b361137f1018c0d330511ebd1/src/Core/src/VisualDiagnostics/VisualDiagnosticsOverlay.iOS.cs#L31

AddObserver returns the dispose method that we then call when we remove the adorners.